src/blockmap.jl

```diff
@@ -483,3 +459,16 @@ end
     end
     return y
 end
+
+Base.@propagate_inbounds function LinearAlgebra.mul!(y::AbstractVector, A::TransposeMap{<:Any,<:BlockDiagonalMap}, x::AbstractVector)
+    mul!(y, transpose(A.lmap), x)
+end
+Base.@propagate_inbounds function LinearAlgebra.mul!(y::AbstractVector, A::AdjointMap{<:Any,<:BlockDiagonalMap}, x::AbstractVector)
+    mul!(y, adjoint(A.lmap), x)
+end
+Base.@propagate_inbounds function LinearAlgebra.mul!(y::AbstractVector, A::TransposeMap{<:Any,<:BlockDiagonalMap}, x::AbstractVector, α::Number, β::Number)
+    mul!(y, transpose(A.lmap), x, α, β)
+end
+Base.@propagate_inbounds function LinearAlgebra.mul!(y::AbstractVector, A::AdjointMap{<:Any,<:BlockDiagonalMap}, x::AbstractVector, α::Number, β::Number)
+    mul!(y, adjoint(A.lmap), x, α, β)
+end
```

Comment:
This pattern of `TransposeMap/AdjointMap{<:Any,<:InvariantMap}` can actually only occur if one wraps an "`InvariantMap`" by a `WrappedMap`, and only then takes the `adjoint`/`transpose`. By `InvariantMap` I mean a `LinearMap` subtype that yields its own type when calling `adjoint` or `transpose` on it. To keep the code cleaner, we could make this the default behavior, and I'll show below where one would do that.